File: 205-isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char,char>mpp1,mpp2;
        for(int i=0;i<s.length();i++){
            if(mpp1[s[i]] && mpp1[s[i]]!=t[i])return false;
            if(mpp2[t[i]] && mpp2[t[i]]!=s[i])return false;
            mpp1[s[i]] = t[i];
            mpp2[t[i]] = s[i]; 
        }
        return true;
    }
};

File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int[] mppS=new int [256];
        int[] mppT=new int[256];
        for(int i=0;i<s.length();i++){
            char cs = s.charAt(i);
            char ct = t.charAt(i);
            if(mppS[cs]!=mppT[ct])return false;
            mppS[cs] = mppT[ct] = i+1;
        }
        return true;
    }
}